libraries/AC_Avoidance/AP_OABendyRuler.cpp

```diff
@@ -0,0 +1,343 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AP_OABendyRuler.h"
+#include <AC_Fence/AC_Fence.h>
+#include <stdio.h>
+
+extern const AP_HAL::HAL &hal;
+
+const int16_t OA_BEARING_INC = 5;               // check every 5 degrees around vehicle
+const float OA_LOOKAHEAD_STEP2_RATIO = 1.0f;    // step2's lookahead length as a ratio of step1's lookahead length
+const float OA_LOOKAHEAD_STEP2_MIN = 2;         // step2 checks at least this many meters past step1's location
+const float OA_LOOKAHEAD_PAST_DEST = 2;         // lookahead length will be at least this many meters past the destination
+const float OA_VERY_BAD_MARGIN = -10000;        // very bad margin to use during initialisation
+const float OA_VERY_GOOD_MARGIN = 10000;        // very good margin to use during initialisation
+
+/// Constructor
+AP_OABendyRuler::AP_OABendyRuler()
+{
+    _singleton = this;
+}
+
+// run background task to find best path and update avoidance_results
+// returns true and updates origin_new and destination_new if a best path has been found
+bool AP_OABendyRuler::update(const Location& current_loc, const Location& destination, const Vector2f &ground_speed_vec, Location &origin_new, Location &destination_new)
+{
+    // exit immediately if neither circular nor polygon fence is enabled
+    const AC_Fence *fence_ptr = AC_Fence::get_singleton();
```

Comment:
thanks.  I'm going to remove this check completely - it was added early on before we added the support for the proximity sensors.

Reply:
AC_Fence isn't required to be around, so you do need the nullptr check.

Reply:
I've also removed the use of fence-ptr's margin in this update() method so it will only be used down in the calc_margin_from_xxxx_fence calls which do include the nullptr check.